src/es/components/molecules/checkoutColorStage/CheckoutColorStage.js

```diff
@@ -0,0 +1,112 @@
+// @ts-check
+import { Shadow } from '../../web-components-toolbox/src/es/components/prototypes/Shadow.js'
+
+/**
+* @export
+* @class Contact
+* @type {CustomElementConstructor}
+*/
+export default class Contact extends Shadow() {
+  constructor (options = {}, ...args) {
+    super({ importMetaUrl: import.meta.url, mode: 'false', ...options }, ...args)
+
+    this.backLinkListener = this.backLinkListener.bind(this)
+  }
+
+  connectedCallback () {
+    if (this.shouldRenderCSS()) this.renderCSS()
+    if (this.shouldRenderHTML()) {
+      this.renderHTML().then(() => {
+        if (!this.backButton) {
+          this.backButton = this.root.querySelector('o-grid').shadowRoot.querySelector('.back-button')
+          this.backButton.addEventListener('click', this.backLinkListener)
+        }
+      })
+    }
+  }
+
+  disconnectedCallback () {
+    this.backButton.removeEvenetListener('click', this.backLinkListener)
+  }
+
+  backLinkListener (event) {
+    event.preventDefault()
+    window.history.back()
+  }
+
+  /**
+   * evaluates if a render is necessary
+   *
+   * @return {boolean}
+   */
+  shouldRenderCSS () {
+    return !this.root.querySelector(`:host > style[_css], ${this.tagName} > style[_css]`)
+  }
+
+  /**
+   * evaluates if a render is necessary
+   *
+   * @return {boolean}
+   */
+  shouldRenderHTML () {
+    return !this.grid
+  }
+
+  /**
+   * renders the css
+   */
+  renderCSS () {
+    this.css = /* css */`
+      :host {
+        width: 100% !important;
+        margin-left: 0 !important;
+        margin-right: 0 !important;
+        background-color: var(--mdx-sys-color-primary-default);
+      }
+    `
+  }
+
+  /**
+   * Render HTML
+   * @returns Promise<void>
+   */
+  renderHTML () {
+    return this.fetchModules([
+      {
+        path: `${this.importMetaUrl}../../web-components-toolbox/src/es/components/organisms/grid/Grid.js`,
+        name: 'o-grid'
+      },
+      {
+        path: `${this.importMetaUrl}../../atoms/heading/Heading.js`,
+        name: 'ks-a-heading'
+      }
+    ]).then(() => {
+      this.html = /* HTML */ `
```

Comment:
you could move the `this.html = ...` to before the return. Then you would have everything rendered in sync. Query selecting the back-button could be done with and without expecting the shadow root of grid to be there.

Reply:
I was trying to do it in synch, but it seems that I can't select the back-button before <o-grid> is initialized. But I might be wrong, just can't get it to work. In case this should be improved, let's synch via teams :) I am available in the afternoon.